react/week3/ToDoList.js

```diff
@@ -0,0 +1,148 @@
+    setCheckedItem({ renderTrigger: !checkedItem.renderTrigger });
+    };
+
+    function taskStyle(checkStatus) {
+    if (checkStatus) {
+      return { textDecoration: "line-through" };
+        };
+    };
+
+    //.............................................................edit section
+
+    const [editItem, setEditItem] = useState({ renderTrigger: false });
+        
+    function editButton(index) {
+
+        //From which index comes the request to edit
+        const indexNumber = mytodolist.findIndex(item => item.id === index);
+        mytodolist[indexNumber].editable = !mytodolist[indexNumber].editable;
+
+        setEditItem({ renderTrigger: !editItem.renderTrigger })
+    };
+    
+    //ASK MENTOR: below function works well without initilizing newdescription state.?
+    const [newDescription, setNewDescription] = useState('');
+
+    function editStyle(index) {
+        
+        const indexNumber = mytodolist.findIndex(item => item.id === index);
+
+        return <div><input
+            type="text"
+            value={mytodolist[indexNumber].description}
+            onChange={(e) => setNewDescription(mytodolist[indexNumber].description = e.target.value)
+        }></input></div>
+    };
+
+    return (
+        <div>
+            <h1>ToDoList</h1>
+            <p>You Have Used {passedTime} Seconds on this website!</p>
+            ToDo description <input
+                type="text"
+                value={description}
+                onChange={changeDescription}></input>
+            <br />
+            deadline <input
+                type="date"
+                value={deadline}
+                onChange={changeDeadline}></input>
+            <br />
+            <button onClick={addToDo}>Add ToDo</button>
+            {mytodolist.length===0 ? <div>There is No Item in the list!</div> : ''}
+            <ul style={{ listStyleType: "none" }}>
+                {mytodolist.map((todo, index) => 
+                    <li key={index}
+                        className="todolistTable"
+                        style={taskStyle(todo.checkStatus)}>
+                        {todo.editable ? editStyle(todo.id) : todo.description } {todo.deadline}
+                        <input
+                            type="checkbox"
+                            onChange={() => changeCheck(todo.id)}
+                            checked={todo.checkStatus} ></input>
+                        <button onClick={() => deleteButton(todo.id)}>Delete</button>
+                        <button onClick={() => editButton(todo.id)}>{todo.editable ? "Update" : "Edit"}</button>
+                    </li>) 
+                }
+            </ul>
+        </div>
+    );
+};
+//ASK MENTOR: when I try to run check just first 3 elements of my list comes the warning as a blow. what that means?
+//index.js:1 Warning: A component is changing an uncontrolled input to be controlled. 
+//This is likely caused by the value changing from undefined to a defined value, 
```

Comment:
I think its because of something you are setting as empty object or array and then late assgning some value  , my suspect is : const [mytodolist, setMytodolist] = useState([]); but I cant tell by looking into code here, Please check stackoverlfow 

https://stackoverflow.com/questions/47012169/a-component-is-changing-an-uncontrolled-input-of-type-text-to-be-controlled-erro